sirepo/uri_router.py

```diff
@@ -190,15 +192,16 @@ def _is_api_func(cls, name, obj):
 
 def start_tornado(ip, port, debug=False):
     """Start tornado server, does not return"""
-    import tornado.httpserver
-    import tornado.ioloop
-    import tornado.web
+    from tornado import httpserver, ioloop, web, log
 
-    class _Handler(tornado.web.RequestHandler):
+    class _Handler(web.RequestHandler):
```

Comment:
Am I right that the reason for using `web.RequestHandler` instead of `tornado.web.RequestHandler` is that it's all inside `start_tornado` which provides the needed context?

Reply:
The reason the names are imported without qualification inside a method is due to the fact that when you say `import tornado.web`, Python actually puts the `tornado` in the global namespace and the import ceases to be local. 

Yes, `_Handler` is only necessary if tornado is being started. It dispatches to uri_router._call_api, which does all the work. The work that has been done over the last many months got us to this point where the adapters for flask and tornado are very small.